src/Type/Constant/ConstantArrayType.php

```diff
@@ -340,38 +342,51 @@ public function equals(Type $type): bool
 
 	public function isCallable(): TrinaryLogic
 	{
-		$typeAndMethod = $this->findTypeAndMethodName();
-		if ($typeAndMethod === null) {
+		$typeAndMethods = $this->findTypeAndMethodNames();
+		if ($typeAndMethods === []) {
 			return TrinaryLogic::createNo();
 		}
 
-		return $typeAndMethod->getCertainty();
+		$results = array_map(
+			static fn (ConstantArrayTypeAndMethod $typeAndMethod): TrinaryLogic => $typeAndMethod->getCertainty(),
+			$typeAndMethods,
+		);
+
+		return TrinaryLogic::createYes()->and(...$results);
 	}
 
 	/**
 	 * @return ParametersAcceptor[]
 	 */
 	public function getCallableParametersAcceptors(ClassMemberAccessAnswerer $scope): array
 	{
-		$typeAndMethodName = $this->findTypeAndMethodName();
-		if ($typeAndMethodName === null) {
+		$typeAndMethodNames = $this->findTypeAndMethodNames();
+		if ($typeAndMethodNames === []) {
 			throw new ShouldNotHappenException();
 		}
 
-		if ($typeAndMethodName->isUnknown() || !$typeAndMethodName->getCertainty()->yes()) {
-			return [new TrivialParametersAcceptor()];
-		}
+		$acceptors = [];
+		foreach ($typeAndMethodNames as $typeAndMethodName) {
+			if ($typeAndMethodName->isUnknown() || !$typeAndMethodName->getCertainty()->yes()) {
+				$acceptors[] = new TrivialParametersAcceptor();
+				continue;
+			}
 
-		$method = $typeAndMethodName->getType()
-			->getMethod($typeAndMethodName->getMethod(), $scope);
+			$method = $typeAndMethodName->getType()
+				->getMethod($typeAndMethodName->getMethod(), $scope);
 
-		if (!$scope->canCallMethod($method)) {
-			return [new InaccessibleMethod($method)];
+			if (!$scope->canCallMethod($method)) {
+				$acceptors[] = new InaccessibleMethod($method);
+				continue;
+			}
+
+			array_push($acceptors, ...$method->getVariants());
 		}
 
-		return $method->getVariants();
+		return $acceptors;
 	}
 
+	/** @deprecated Use findTypeAndMethodNames() instead  */
 	public function findTypeAndMethodName(): ?ConstantArrayTypeAndMethod
```

Comment:
not sure what the best approach here is. I wanted to play it safe and don't touch it at all. But `findTypeAndMethodNames` is basically a super type of this one. What I want to say - this most likely could be replaced by smth like
```php
$typeAndMethodNames = $this->findTypeAndMethodNames();
return count($typeAndMethodNames) === 1 ? $typeAndMethodNames[0] : null;
```